src/ol/layer/vectortile.js

```diff
@@ -34,17 +34,23 @@ ol.layer.VectorTile = function(opt_options) {
   this.setUseInterimTilesOnError(options.useInterimTilesOnError ?
     options.useInterimTilesOnError : true);
 
-  ol.asserts.assert(options.renderMode == undefined ||
-      options.renderMode == ol.layer.VectorTileRenderType.IMAGE ||
-      options.renderMode == ol.layer.VectorTileRenderType.HYBRID ||
-      options.renderMode == ol.layer.VectorTileRenderType.VECTOR,
+  var renderMode = options.renderMode;
+
+  ol.asserts.assert(renderMode == undefined ||
+      renderMode == ol.layer.VectorTileRenderType.IMAGE ||
+      renderMode == ol.layer.VectorTileRenderType.HYBRID ||
+      renderMode == ol.layer.VectorTileRenderType.VECTOR,
   28); // `renderMode` must be `'image'`, `'hybrid'` or `'vector'`
 
+  if (options.declutter && renderMode == ol.layer.VectorTileRenderType.IMAGE) {
+    renderMode = ol.layer.VectorTileRenderType.HYBRID;
```

Comment:
Can we somehow notify users that we change the rendering mode?

Reply:
It's documented [here](https://github.com/openlayers/openlayers/pull/7328/files#diff-044fc1d937cdd730e52ddad85740de2fR4412). Were you thinking of a more prominent place? I don't think anyone has ever used the`'image'` mode.

Reply:
I was thinking about logging to the console, but as we don't do that anywhere else the documentation you linked to is enough.